src/refactor_nrepl/artifacts.clj

```diff
@@ -10,10 +10,24 @@
             [refactor-nrepl.ns.slam.hound.regrow :as slamhound-regrow]
             [version-clj.core :as versions])
   (:import java.util.Date
+           java.util.zip.GZIPInputStream
            java.util.jar.JarFile))
 
+(def artifacts-file (str (System/getProperty "java.io.tmpdir")
+                         "/refactor-nrepl-artifacts-cache"))
+
+(defn get-last-modified-from-file
+  "Returns last modified time in milliseconds or nil if file does not exist."
+  [file]
+  (let [lm (.lastModified (io/file file))]
+    (if (zero? lm) nil lm)))
+
 ;;  structure here is {"prismatic/schem" ["0.1.1" "0.2.0" ...]}
-(defonce artifacts (atom {} :meta {:last-modified nil}))
+(defonce artifacts (atom (if (.exists (io/as-file artifacts-file))
+                           (->> artifacts-file slurp read-string (into (sorted-map)))
```

Comment:
We should be able to use `edn/read-string` as a drop-in replacement here and avoid exposing the full reader like this.  As you probably know `read-string` is capable of evaluating any code inserted into that file.

Reply:
Haha my spidy senses tingled on that but for some reason I never really understood that that's one of the things edn brought to the table. Thanks!

Reply:
You could also bind [\*read-eval\*](https://clojuredocs.org/clojure.core/*read-eval*) to `false`, but it reads way better to just use the EDN reader when possible :)